saleor/order/utils.py

```diff
@@ -273,11 +275,16 @@ def create_order_line(
     unit_discount = line.undiscounted_unit_price - line.unit_price
     if unit_discount.gross:
         if rules_info:
-            line_discounts = create_order_line_discounts(line, rules_info)
+            # TODO zedzior: this trigger multiple db calls for multiple lines
+            line_discounts = (
+                create_order_line_discount_objects_for_catalogue_promotions(
+                    line, rules_info, channel
+                )
+            )
             promotion = rules_info[0].promotion
             line.sale_id = get_sale_id(promotion)
             line.unit_discount_reason = (
-                prepare_promotion_discount_reason(promotion, line.sale_id)
+                prepare_promotion_discount_reason(rules_info[0].rule)
```

Comment:
Isn't like this will trigger additional DB call?
`prepare_promotion_discount_reason` calls `rule.promotion`. `rule` i `PromotionRule` object. You already have the promotion inside `rules_info[0].promotion`

Reply:
Good point